sdcm/nemesis.py

```diff
@@ -2324,7 +2324,18 @@ def remove_node():
             # the entire nemesis.
 
             exit_status = remove_node()
-            assert exit_status == 0, "nodetool removenode command exited with status {}".format(exit_status)
+
+            # if remove node command failed by any reason,
+            # we will remove the terminated node from
+            # dead_nodes_list, so the health validator terminate the job
+            if exit_status != 0:
+                self.log.error(f"nodetool removenode command exited with status {exit_status}")
+                self.log.debug(f"Remove failed node{node_to_remove} from dead node list {self.cluster.dead_nodes_list}")
+                node = next((n for n in self.cluster.dead_nodes_list if n.ip_address == node_to_remove.ip_address), None)
```

Comment:
why are you using `next()`?

Reply:
(n for n in self.cluster.dead_nodes_list if n.ip_address == node_to_remove.ip_address) - it is generator. So next() i use to return the value from generator or return none